src/screens/Exchange/ExchangeConfirm.js

```diff
@@ -134,146 +112,78 @@ const Row = styled.View`
-      contractAddress,
-      data,
-      symbol: fromAssetSymbol,
-    } = this.transactionPayload;
-    const value = Number(amount || 0);
+// already passed as mixed from other screen, TODO: fix the whole flow?
+type MixedOfferOrder = $Shape<OfferOrder & TokenTransactionPayload>;
 
-    const isConvertedTx = isWethConvertedTx(fromAssetSymbol, contractAddress);
+const ExchangeConfirmScreen = ({
+  executingExchangeTransaction,
+  navigation,
+  balances,
+  baseFiatCurrency,
+  rates,
+  theme,
+  isOnline,
+  setDismissTransaction,
+  feeInfo,
+  isEstimating,
+  estimateErrorMessage,
+  estimateTransaction,
+  accountAssets,
+  supportedAssets,
+}: Props) => {
+  useEffect(() => {
+    if (!executingExchangeTransaction) navigation.goBack();
+  }, [executingExchangeTransaction]);
+
+  const offerOrder: MixedOfferOrder = navigation.getParam('offerOrder', {});
+
+  const {
+    fromAsset,
+    toAsset,
+    setTokenAllowance,
+    provider,
+    receiveQuantity,
+    payQuantity,
+    decimals,
+    amount,
+    symbol,
+    to: recipient,
+    contractAddress,
+    data,
+  } = offerOrder;
+
+  const fetchTransactionEstimate = () => {
+    const isConvertedTx = isWethConvertedTx(symbol, contractAddress);
 
     // for WETH converted txs on homestead, we need to provide ETH data or else estimation is always 0$
     const contractAddressForEstimation = isProdEnv && isConvertedTx
       ? ethersConstants.AddressZero
       : contractAddress;
 
-    const { symbol, decimals } =
-      getAssetDataByAddress(getAssetsAsList(accountAssets), supportedAssets, contractAddressForEstimation);
-    const assetData = { contractAddress: contractAddressForEstimation, token: symbol, decimals };
-
-    let transaction = { recipient, value };
-    if (data) transaction = { ...transaction, data };
-
-    const estimated = await smartWalletService
-      .estimateAccountTransaction(transaction, assetData)
-      .then(result => buildTxFeeInfo(result, useGasToken))
-      .catch(() => null);
+    const estimateAsset = getAssetDataByAddress(
+      getAssetsAsList(accountAssets),
+      supportedAssets,
+      contractAddressForEstimation,
+    );
 
-    if (!estimated) {
-      return defaultResponse;
-    }
+    const estimateAssetData = {
+      contractAddress: estimateAsset.address,
+      token: estimateAsset.symbol,
+      decimals: estimateAsset.decimals,
+    };
 
-    return estimated;
+    estimateTransaction(recipient, Number(amount || 0), data, estimateAssetData);
   };
 
-  getKeyWalletTxFee = (txSpeed?: string, gasLimit?: number): TransactionFeeInfo => {
-    const { gasInfo } = this.props;
-    txSpeed = txSpeed || SPEED_TYPES.NORMAL;
-    gasLimit = gasLimit || this.state.gasLimit || 0;
+  useEffect(() => { fetchTransactionEstimate(); }, []);
 
-    const gasPrice = gasInfo.gasPrice[txSpeed] || 0;
-    const gasPriceWei = utils.parseUnits(gasPrice.toString(), 'gwei');
+  const onConfirmTransactionPress = () => {
+    if (!feeInfo) return;
```

Comment:
How does this appear to the user? Is it like a no-op?

Reply:
Actually I've covered this scenario in other screens as Toast message, will apply here as well, thanks!